TheJumpCloud/support
# Sa 2336 update jc template import

## Issues
* [SA-2336](https://jumpcloud.atlassian.net/browse/SA-2336) - Import/Update-JCUsersFromCSV & New-JCImportTemplate: alternateEmail, manager, managedAppleID attributes
* [SA-2330](https://jumpcloud.atlassian.net/browse/SA-2330) - New/Get/Set-JCUser: alternateEmail, manager, managedAppleID attributes

## What does this solve?

The alternateEmail, manager & managedAppleID fields were missing from the New/Get/Set-JCUser, Import/Update-JCUsersFromCSV & New-JCImportTemplate functions - this release adds those fields to those functions. 

## Is there anything particularly tricky?

The manager field is a bit interesting. Keeping in line with how we've set other functions in the JCModule, the manager field will accept a string input of either an ID or a username. If a username is provided the function will do a lookup and determine the correct manager ID, if an ID is provided the function will add the manager field to the user object. 

## How should this be tested?

Get-JCUser:
- Test that `Get-JCUser -username "someUser"` should return all alternateEmail, manager, managedAppleID attributes 
- Test that `Get-JCUser -userid "someUserid"` should return all alternateEmail, manager, managedAppleID attributes 
Set-JCUser:
- Test that `Set-JCUser -username "someUser"` can set all alternateEmail, manager, managedAppleID attributes 
- Test that `Set-JCUser -userid "someUserid" can set all alternateEmail, manager, managedAppleID attributes 
New-JCuser:
- Test that running New-JCUser can add a user and set the alternateEmail, manager, managedAppleID attributes 
New-JCImportTemplate
- Running this function should produce a csv with the alternateEmail, manager, managedAppleID attributes 
Import-JCUserFromCSV
- New users can be added with alternateEmail, manager, managedAppleID attributes 
Update-JCUserFromCSV
- Existing users can be updated and their alternateEmail, manager, managedAppleID attributes can be set

## Screenshots
